src/services/data_model_engine.js

```diff
@@ -255,7 +255,7 @@ export default class DataModelEngine {
       throw new Error(`Could not fetch the bundle from the shelterer: ${err.message || err}`);
     }
 
-    await this.bundleRepository.setBundleRepository(bundleId, BundleStatuses.downloaded, {nodeUrl});
+    await this.bundleRepository.setBundleRepository(bundleId, BundleStatuses.downloaded, {nodeUrl, holdUntil: new Date(challengeExpirationTime)});
```

Comment:
Why do we store `holdUntil` in two places? `metadata` and `repository`.
After successful resolution we update it only in `repository` and leave outdated in `metadata`.

Reply:
:+1: Added migration